downstairs/src/lib.rs

```diff
@@ -3008,21 +3037,15 @@ impl Downstairs {
         Ok(())
     }
 
-    async fn on_negotiation_step<W>(
+    async fn on_negotiation_step(
         &mut self,
         m: Message,
         state: &mut NegotiationData,
-        fw: &mut crucible_protocol::MessageWriter<W>,
-    ) -> Result<()>
-    where
-        W: tokio::io::AsyncWrite
-            + std::marker::Unpin
-            + std::marker::Send
-            + 'static,
-    {
+        fw: &mut mpsc::UnboundedSender<Message>,
```

Comment:
Could re rename this to `reply_channel_tx` to match what the caller calls the channel?

Reply:
done!